Step : 1
Checked the dataset Manually once to make sure that the column name makes sense and looked at the Data Dictionary to understand the terminologies.
The dataset is in CSV format.

In [110]:
import pandas as pd               # Pandas is a package which has ability to handle dataframes
import numpy as np               # Numpy has an ability to handle arrays
import matplotlib.pyplot as plt    # mathplotlib for plot
import seaborn as sns   # box plot
loan = pd.read_csv("/content/loan.csv")    #Loan is a Dataframe variable ,
                                           #with the help of pandas, csv data is loaded into the variable "loan" using read_csv method

<ipython-input-110-cec56c3c7a97>:5: DtypeWarning: Columns (47) have mixed types. Specify dtype option on import or set low_memory=False.
  loan = pd.read_csv("/content/loan.csv")    #Loan is a Dataframe variable ,




* Check the dimension of the dataset
* Understand how many of them are categorical and how many of them are Numerical



In [108]:
loan.shape     # check the dimension of the dataset  - (rows, columns)

(39717, 111)

The Number of rows are 39717 and columns are 111

In [ ]:
print(loan.isna().sum())    # check the sum of na values columnwise

In [90]:
loan = loan.loc[:, loan.isin([' ','NULL',0]).mean() < .7]      # retains only the columns which have less than 60% of na value.

In [91]:
loan = loan.drop_duplicates()  # Remove duplicates if any  - optional code still good to test if it has duplicates

In [93]:
drop_cols = [c for c
             in list(loan)
             if loan[c].nunique() <= 1]    # collects the name of columns which has less than 1 unique or distinct values
loan = loan.drop(columns=drop_cols)        # drops the collected columns in previous step

A column with a same value for all the rows may not contribute to result.
So we removed such columns in the above code snippet.

In [94]:
extra_cols = ['id','member_id','next_pymnt_d','title',
'last_pymnt_d',
'issue_d',
'url',
'desc',
'zip_code', # Address state code
'earliest_cr_line',
'last_credit_pull_d','emp_title' ]           # for this analysis, unnecessary column like dates and url, description are considered as irrelevant
loan =loan.drop(extra_cols,axis=1)  # all irrelevant columns are removed.

For the simple analysis of to saction  load, date columns may not contribute more. Still we can gain the difference between the dates and duration may be considered as one of factors.
But for this analysis, we removed all the date columns and columns provide extra information like url, description.  we tend to remove zip code as we already have address state code.

In [ ]:
print(loan.describe) # Overview of dataset

In [96]:
print(loan.columns)

Index(['loan_amnt', 'funded_amnt', 'funded_amnt_inv', 'term', 'int_rate',
       'installment', 'grade', 'sub_grade', 'emp_length', 'home_ownership',
       'annual_inc', 'verification_status', 'loan_status', 'purpose',
       'addr_state', 'dti', 'inq_last_6mths', 'mths_since_last_delinq',
       'mths_since_last_record', 'open_acc', 'revol_bal', 'revol_util',
       'total_acc', 'total_pymnt', 'total_pymnt_inv', 'total_rec_prncp',
       'total_rec_int', 'last_pymnt_amnt'],
      dtype='object')


There are only 28 remaining useful columns for analysis.
out of these 28 fields, annual income field has outliers so top and bottom 5% of the values are removed

In [ ]:
upper_bound = np.percentile(loan['annual_inc'], 95)
lower_bound = np.percentile(loan['annual_inc'], 5)

# Remove the outliers
loan = loan[(loan['annual_inc'] <= upper_bound) & (loan['annual_inc'] >= lower_bound)]
loan

In [102]:
loan.to_csv('filename.csv') # we can analyse it in excel using this exported file

In [ ]:
# Store the list of columns
columns_to_plot = ['loan_amnt', 'funded_amnt']

# Create the figure and two subplots
fig, axes = plt.subplots(ncols=len(columns_to_plot ))

# Create the boxplot with Seaborn
for column, axis in zip(columns_to_plot , axes):
        sns.boxplot(data=loan[column], ax=axis)
        axis.set_title(column)
        # axis.set(xticklabels=[], xticks=[], ylabel=column)

# Show the plot
plt.tight_layout()
plt.show()

In [ ]:
plt.scatter(loan['loan_amnt'], loan['funded_amnt'], color='red')

# Show the plot
plt.show()